src/MUIDataTable.js

```diff
@@ -1229,6 +1233,10 @@ class MUIDataTable extends React.Component {
         break;
     }
 
+    let tableProps = this.options.setTableProps ? this.options.setTableProps() : {};
```

Comment:
Maybe we should be more defensive here. In case the function doesn't return an object, it will error out below. What do you think?

Reply:
I don't know, that's not how the existing setCellProps and setRowProps functions work. Trying out other return values in the console, it doesn't look like anything breaks with the spread operator, they just wouldn't be setting any fields if they had the function return a non-object (with the exception of a "string" type, where they'd set numerical fields on the component).

Reply:
The others don't work this way because they aren't checking an object for a specific property. The new functions you added are checking to see if `classNames` appears on the object, and this will error for non-objects, where the others won't, with the exception of strings.